### ISSUE TITLE
Postman VS Code extension should allow import/export to OpenAPI and Postman JSON

### ISSUE THREAD

**Issue by codemaestro** [a85/POSTMan-Chrome-Extension#11728]
### Is there an existing request for this feature?

- [X] I have searched the existing issues for this feature request and I know that duplicates will be closed

### Is your feature request related to a problem?

This is a feature request against the preliminary implementation announced in a recent email and by an issue logged Apr 2021: #9890

### Describe the solution you'd like

Based on the screenshot of the the proposed VS Code extension, the extension is expected to provide a Postman client/interface within VS Code, which would be really useful. We use the Postman desktop/web client heavily to organize the structure of our collections for use with our API documentation. Our documentation is stored in separate text files on the local machine, and a CI/CD process merges the collection JSON file with the text files.

We would need to export different versions of the collection:

1. The collection as is. One of our issues is that our writers sometimes forget to export the latest copy of the collection before pushing a merge request back to our repo. The VS Code extension would make it easier to remember to download the latest copy of the collection before merge. It might even help to include the download as a pre-commit hook.

2. A copy of the collection piped through our merge script. We would write a custom script that extracts the portion of our CI/CD pipeline that merges the collection with the local files, and export the Postman JSON with the documentation intact.

3. A copy of the collection in (2) above piped to an OpenAPI specification. We would write a script template that contains the OpenAPI structure we would then use for integrating into other tools.

4. Items (2) and (3) suggest that the export routine be customizable.

### Describe alternatives you've considered

Our existing workflow already performs some elements of this in our CI/CD pipeline. It's just that exporting the collection is always a manual step before pushing to our release branch.

**Comment by preethammavin** [comment]
@codemaestro Looks like an interesting workflow, curious to know where the documentation powered by Postman collection is published. Even with your current workflow, you should be able to automate most things with Postman Workspaces, Postman Public API, and Version Control in Postman.

Would be great if we can get on a call to understand your workflow better. Here's my [calendly link](https://calendly.com/d/2bt-vk9-yjh/one-off-meeting), if these time slots dont workdo drop me an email at preetham@postman.com so we can set some time up

**Comment by codemaestro** [comment]
Hi @preethammavin! I CC:ed you on an open ticket with your support team to track additional feedback outside of GitHub. I'll coordinate a meeting, thanks.

**Comment by thena-seer-sfg** [comment]
Yes, we commit exported Postman collection JSON files to our repo, and it would be nice to be able to import them easily into the extension from files in the workspace, and re-export after making changes, or even better be able to live-edit files using the postman interface. The curl feature doesn't seem to support `file://` protocol.

**Comment by mees-van-wel** [comment]
This would indeed be a great feature!

**Comment by jayendra-ram** [comment]
i need this so much

**Comment by paulrwest** [comment]
We need this too, we have a team working on postman collections as a team under version control live-edit would be the ideal solution import export to file second best.

**Comment by pankajk07** [comment]
Hey all, we have added the support to import Postman Collections and Environments into the extension. This feature is available as part of the latest update. 

The following video shows how we can import a collection inside the extension.

https://github.com/postmanlabs/postman-app-support/assets/44100113/1f5b428d-96db-42ee-911e-563789d0b521

**Comment by slukashuk** [comment]
So for now only importing of collection available? I can't find how can I export my collection

**Comment by slukashuk** [comment]
And during importing a collection if duplicate found - there is no option to replace the file, only importing as a copy or cancel 
![image](https://github.com/postmanlabs/postman-app-support/assets/5691714/fbdbac9e-fd15-4e3c-b437-743fb0cb8b7c)
Replace option can make usage of the extension much easier 
![image](https://github.com/postmanlabs/postman-app-support/assets/5691714/a0844c08-6c32-4ae5-a7e8-8d8bd449abbe)

**Comment by vanjis** [comment]
Looking for an export collection feature.

**Comment by tony-ross** [comment]
export collection is a must - our wow needs this feature.

**Comment by dongokovacs** [comment]
Can anybody import json collection? I get only a loading animation

**Comment by indraneel-postman** [comment]
@daniavander Can you share some more details on what issue you are facing when importing a collection? The below information would be helpful:
- Video recording of the issue
- Version of Postman Extension
- Extension Log (from the three dots menu)

Would request you to create a new issue for this if you are consistently facing it while importing files.

**Comment by tmolumby** [comment]
I looks like there is a workaround for exporting the JSON of a collection in the VSCode extension:

1. Go to https://web.postman.co
2. Go to your profile
3. Goto collections
4. Now you have the much more robust web interface and you can now right click on the collection and export

Would be nice if this functionality was in the extension.

**Comment by joao-rebelo** [comment]
Having to go to the website kind of defeats the purpose of having all locally at the IDE. We need the local workflow import/edit/export locally to integrate with remaining tooling.
Even better would be to work "offline" on a collection stored locally at the disk, so that a "save" would be stored at the collection file directly.

**Comment by alicolakmhp** [comment]
it would be nice to have the export functionality in the VsCode extention
